crates/sui-framework/packages/sui-system/sources/sui_system_state_inner.move

```diff
@@ -865,18 +865,29 @@ module sui_system::sui_system_state_inner {
 
         let storage_charge = storage_reward.value();
         let computation_charge = computation_reward.value();
-
+        let mut stake_subsidy = balance::zero();
         // Include stake subsidy in the rewards given out to validators and stakers.
         // Delay distributing any stake subsidies until after `stake_subsidy_start_epoch`.
         // And if this epoch is shorter than the regular epoch duration, don't distribute any stake subsidy.
-        let stake_subsidy =
-            if (ctx.epoch() >= self.parameters.stake_subsidy_start_epoch  &&
-                epoch_start_timestamp_ms >= prev_epoch_start_timestamp + self.parameters.epoch_duration_ms)
-            {
-                self.stake_subsidy.advance_epoch()
-            } else {
-                balance::zero()
+        if (ctx.epoch() >= self.parameters.stake_subsidy_start_epoch  &&
+            epoch_start_timestamp_ms >= prev_epoch_start_timestamp + self.parameters.epoch_duration_ms)
+        {
+            // special case for epoch 560 -> 561 change bug. add extra subsidies for "safe mode"
+            // where reward distribution was skipped. use distribution counter to avoiding affecting devnet
+            // and testnet
+            if (self.stake_subsidy.get_distribution_counter() == 540) {
+                let first_safe_mode_epoch = 561;
```

Comment:
Not fully certain, but I believe this value should be `560` if we're comparing it against `ctx.epoch()`. Context contains current epoch (not the one we're moving to), so in case of `561 -> 562` context would contain 561, and the code won't execute. It will run in `562 -> 563` epoch, because Context would be 562.

In other words, if we want to run it from the missed epoch, we need to either use `new_epoch` as the value we're comparing to (not `ctx.epoch()`). Or, if we decide to check against current epoch, the value should be set to `560`.

Reply:
Nvm. After writing a test I see that it catches up at least once anyway.